src/Microsoft.ML.Data/Transforms/Hashing.cs

```diff
@@ -599,8 +599,7 @@ public uint HashCore(uint seed, uint mask, in double value)
                 ulong v = FloatUtils.GetBits(value == 0 ? 0 : value);
                 var hash = Hashing.MurmurRound(seed, Utils.GetLo(v));
                 var hi = Utils.GetHi(v);
-                if (hi != 0)
-                    hash = Hashing.MurmurRound(hash, hi);
+                hash = Hashing.MurmurRound(hash, hi);
```

Comment:
>hash = Hashing.MurmurRound(hash, hi); [](start = 16, length = 37)

If you change the hashing function, you need to bump the version of the model, while maintaining backwards compatibility.

Reply:
If there isn't a unit test for backwards compatibility: could add one which loads an old model containing a `CategoricalHash` transform, runs a row of data through it, and checks for equality on its output (from the `CateoricalHash`).

Reply:
Agree. Since this is a draft PR, I just modified the existing test cases and wait for ort 1.3 release. I'll add a separate test specifically for testing new types once it's time to bring this PR official

Reply:
synced with Harish that the change will be made just on V2 hash function for 64bit(not released for V2 so backward compatibility is not required this time)

Reply:
I think it's better to be on the safe side, since users can theoretically use nightly versions (or build their own version of ML.NET from the source code), and create models using a version from between releases. 

Either way, the original implementations of `IHasher` (those that don't have "V2" in their name) should not change.

---
In reply to: [425509011](https://github.com/dotnet/machinelearning/pull/5104#discussion_r425509011) [](ancestors = 425509011)

Reply:
The nightly versions are the "bleeding edge". We do not make any supportability guarantees for the nightly versions. 

---
In reply to: [425607553](https://github.com/dotnet/machinelearning/pull/5104#discussion_r425607553) [](ancestors = 425607553,425509011)

Reply:
> I think it's better to be on the safe side, since users can theoretically use nightly versions (or build their own version of ML.NET from the source code), and create models using a version from between releases.
> 
> Either way, the original implementations of `IHasher` (those that don't have "V2" in their name) should not change.
> 
> In reply to: [425509011](https://github.com/dotnet/machinelearning/pull/5104#discussion_r425509011) [](ancestors = 425509011)

Those non-V2s have been reverted to original